### ISSUE TITLE
Moonlight Steamlink:  Graphics glitch, intermittent grey triangle overlay with higher resolutions/bitrates

### ISSUE THREAD

**Issue by geekycow** [moonlight-stream/moonlight-qt#254]
**Describe the bug**
Streaming display has a grey triangle in the top right of the screen in about 1 in 4 instances of firing up the link when used at 1080p/60Hz 30Mbps or higher.

**Steps to reproduce**
- Select Moonlight starting with default settings.
- Change resolution to 1080p/60Hz and 30Mbps or higher.
- Connect to streaming host.
- Select Steam.
- If a grey triangle doesn't appear, stop streaming from within Steam power menu and select Steam again, repeat until grey triangle appears.

**Screenshots**
See attached.

**Affected games**
All apps, including Steam, whether or not Steam is running.

**Moonlight settings (please complete the following information)**
- Have any settings been adjusted from defaults?
Yes.

- If so, which settings have been changed?
Resolution and bitrate have been increased to 1080p/60Hz / 30Mbps.

- Does the problem still occur after reverting settings back to default?
No.  Doesn't seem to happen at the default resolution and bitrate of 720p/60Hz 10Mbps.

**Client PC details (please complete the following information)**
- Device: Steam Link
- OS: 692 (public)
- Network: 1Gbps
- Moonlight Version: moonlight steam link v1.0.1

**Server PC details (please complete the following information)**
- OS: Windows 10 Pro 17134 64bit
- CPU: Intel i5-4460
- GPU: GTX 960 2Gb
- RAM: 32Gb
- Network: 1Gbps
- GeForce Experience version: 3.19.0.94
- Nvidia GPU driver: 417.01

**Moonlight Logs (please attach)**
Attached.

**Additional context**
Triangle gets larger and issue seems to become more frequent the higher the bitrate.

![moonlight-steamlink-v101-greytriangle](https://user-images.githubusercontent.com/16445944/60619978-de036180-9dd1-11e9-8b8b-ec045fe00f65.jpg)
[moonlight.log](https://github.com/moonlight-stream/moonlight-qt/files/3356597/moonlight.log)

**Comment by cgutman** [comment]
Yep, this is known and documented on the Steam Link community forum post. I suspect a bug in the Vivante GPU drivers.

**Comment by geekycow** [comment]
Sorry.  I didn't realise it was reported on other forums already.
Fortunately it's a very minor issue anyway.

Thank you SO much for your hard work on this project.  :)

**Comment by Forceflow** [comment]
I've also experienced this.
I've captured a moonlight log from a session where the corruption appeared:

[moonlight.log](https://github.com/moonlight-stream/moonlight-qt/files/3893642/moonlight.log)
![moonlight_settings](https://user-images.githubusercontent.com/3607063/69664980-b5824980-1089-11ea-8094-568dfedc4625.jpg)
![moonlight_corruption](https://user-images.githubusercontent.com/3607063/69664988-b7e4a380-1089-11ea-9da3-04cab68e3131.jpg)

**Comment by Forceflow** [comment]
Reply from @slouken (Valve) in mail convo: 

I haven't seen  that corruption before. Are you respecting the 30 Mbit limit on the video stream? If you exceed that you might be running into hardware limitations.

Looking at the bug it seems like this is the case. Moonlight should limit the bitrate to 30 Mbit maximum on the Steam Link hardware.

**Comment by cvandermeer** [comment]
I have the same problem

**Comment by Forceflow** [comment]
I've also systematically lowered the video bitrate every time I've encountered this. I'm on 26.5 atm.

**Comment by mattbell87** [comment]
+1 watching this issue

**Comment by cvandermeer** [comment]
After restarting i do not have this issue anymore, seems to be sporadic?

**Comment by Forceflow** [comment]
> After restarting i do not have this issue anymore, seems to be sporadic?
I'm afraid I also get this after a fresh boot sometimes. Every time I have it, I lower the video bitrate by 0.5. I'm on 24.5 now, starting to think it is - unfortunately - not connected to bitrate.

Anyone else experiencing this on really low bitrates?

**Comment by Forceflow** [comment]
Quick update on the workaround for this: I find it easier if you configure moonlight for steam link to not close the app when you disconnect from the session. That's an optional feature that is in the options screen in the part that is _below the screen output_ usually - so quickly attach a mouse to your Steam link configure that.

You either get the grey corruption in the stream, or you don't. It doesn't pop up halfway through streaming. So a quick stream disconnect (start, select, left shoulder, right shoulder) and resume usually fixes it. Your game (in my case: RDR2) can keep on loading and doing it's thing in the background.

**Comment by franciscomoma** [comment]
+1 With this problem with default configuration. My computer is a GTX 1050 + 7th gen Intel Core i7 running the latest update of Windows 10

**Comment by dryfish** [comment]
I've just started seeing this after playing fine with default moonlight settings (720p60?) for a week or so. The SteamLink performed a software update yesterday and I've been getting the grey triangle ever since (mostly).

Last night I swapped between moonlight and steam streaming then back to moonlight but at 1080p60 and the grey triangle went away.

This morning it's back again (still at 1080p60) - no time to play at the moment but have just managed to ssh into steamlink.
